test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -81,6 +97,27 @@ class test_ucp_sockaddr : public ucp_test {
         EXPECT_EQ(send_data, recv_data);
     }
 
+    void connect_and_send_recv(struct sockaddr *connect_addr)
+    {
+        ucp_ep_params_t ep_params = ucp_test::get_ep_params();
+        ep_params.field_mask      |= UCP_EP_PARAM_FIELD_FLAGS |
+                                     UCP_EP_PARAM_FIELD_SOCK_ADDR |
+                                     UCP_EP_PARAM_FIELD_ERR_HANDLING_MODE;
+        ep_params.err_mode         = UCP_ERR_HANDLING_MODE_PEER;
+        ep_params.flags            = UCP_EP_PARAMS_FLAGS_CLIENT_SERVER;
+        ep_params.sockaddr.addr    = connect_addr;
+        ep_params.sockaddr.addrlen = sizeof(*connect_addr);
+        sender().connect(&receiver(), ep_params);
+
+        tag_send(sender(), receiver());
+
+        /* wait for reverse ep to appear */
+        while (receiver().get_num_eps() == 0) {
+            progress();
+        }
+        tag_send(receiver(), sender());
```

Comment:
hmm.... i don't see paired recvs, but function is named `send_recv`. Looks like the data will be dropped?
Also, how about to send response directly from callback on server side instead of `get_num_eps` check above? I think it can be popular case.

Reply:
the tag_send function has a call to ucp_tag_recv_nb inside it.
it's not really sending a response, it's just making sure that both client and server sides can send/recv from one to another after the connection between them was established.

Reply:
1) ok, then i would suggest to rename `tag_send` to `tag_send_recv`
2) is there any test if EP is able to send directly from CB on server side?

Reply:
1. ok i will rename.
2. no.